language-support/ts/daml-ledger/index.ts

```diff
@@ -953,6 +970,51 @@ class Ledger {
     return decode(partyInfoDecoder, json);
   }
 
+  /**
+   * Fetch a list of all package IDs from the ledger.
+   *
+   * @returns List of package IDs.
+   *
+   */
+  async listPackages(): Promise<PackageId[]> {
+    const json = await this.submit('v1/packages', undefined, 'get');
+    return decode(jtv.array(jtv.string()), json);
+  }
+
+  /**
+   * Fetch a binary package.
+   *
+   * @returns The content of the package as a raw ArrayBuffer.
+   *
+   */
+  async getPackage(id: PackageId): Promise<ArrayBuffer> {
+    const httpResponse = await fetch(this.httpBaseUrl + 'v1/packages/' + id, {
+      headers: this.auth(),
+      method: 'get',
+    });
+    await this.throwOnError(httpResponse);
+    return await httpResponse.arrayBuffer();
+  }
+
+  /**
+   * Upload a binary archive. Note that this requires admin privileges.
+   *
+   * @returns True on success; throws on error.
+   *
+   */
+  async uploadDarFile(abuf: ArrayBuffer): Promise<true> {
```

Comment:
```suggestion
  async uploadDarFile(abuf: ArrayBuffer): Promise<{}> {
```

Reply:
I don't really have an opinion here. `true` made sense in the pseudo-Either case because then you could write `if (uploadDar)`, but in the throwing case I don't think any return value particularly makes sense.

Perhaps the sensible thing to do then would be to return `undefined`? (i.e. just `return;`)

Reply:
looking around a bit `Promise<void>` seems pretty common to the point where the typings specialcase this such that `.resolve()` is typed correctly for a void promise so maybe that?

Reply:
Switched to `Promise<void>`.